MBBSEmu/HostProcess/ExportedModules/Majorbbs.cs

```diff
@@ -8084,5 +8091,117 @@ private void gettime()
 
             Module.Memory.SetArray(timePointer, dosTimeStruct.Data);
         }
+
+        /// <summary>
+        ///     Full Screen Editor
+        ///
+        ///     We re-use the FSD, using a pre-defined custom template, field spec, and formats (in Assets Folder)
+        /// </summary>
+        private void bgnedt()
+        {
+            var textLength = GetParameter(0);
+            var textPointer = GetParameterPointer(1);
+            var topicLength = GetParameter(3);
+            var topicPointer = GetParameterPointer(4);
+            var onDoneEditing = GetParameterPointer(6);
+            var flags = GetParameter(8);
+
+            //This is where we'll build the Answers String to pass into the FSD Template
+            var answersString = new StringBuilder(topicLength + textLength);
+
+            //Compile a Topic Answer String
+            answersString.Append($"TOPIC={Encoding.ASCII.GetString(Module.Memory.GetString(topicPointer))}");
+            
+            //Take the input text and split it out on new lines to we can hydrate the individual lines on the template
+            var rawText = Encoding.ASCII.GetString(Module.Memory.GetString(textPointer, true));
+            var split = rawText.Split('\r');
+            var lineNumber = 0;
+            for (var i = 0; i < split.Length; i++)
+            {
+                if (split[i].Length < 79)
+                {
+                    answersString.Append($"LINE{i}={split[i]}\0");
+                    lineNumber++;
+                    continue;
+                }
+
+                //Too much data for one line, we need to "word wrap."
+                while (split[i].Length > 79)
+                {
+                    answersString.Append($" LINE{lineNumber}={split[i].Substring(0, 79)}\0");
+                    split[i] = split[i][79..];
+                    lineNumber++;
+                }
+            }
+
+            //Double Null Terminated
+            answersString.Append('\0');
+
+            //FSDROOM
+            var resourceManager = new ResourceManager();
+
+            var template = resourceManager.GetResource("MBBSEmu.Assets.fseTemplate.ans");
+            var fieldSpec = resourceManager.GetResource("MBBSEmu.Assets.fseFieldSpec.txt");
+
+            var fsdBufferPointer = Module.Memory.GetOrAllocateVariablePointer($"FSD-TemplateBuffer-{ChannelNumber}", 0x2000);
+            var fsdFieldSpecPointer = Module.Memory.GetOrAllocateVariablePointer($"FSD-FieldSpec-{ChannelNumber}", 0x2000);
+
+            //Zero out FSD Memory Areas
+            Module.Memory.SetZero(fsdBufferPointer, 0x2000);
+            Module.Memory.SetZero(fsdFieldSpecPointer, 0x2000);
+
+            //Hydrate FSD Memory Areas with Values
+            Module.Memory.SetArray(fsdBufferPointer, template);
```

Comment:
+ null terminators?

Reply:
We set zero prior to the areas declares and then fill them in. So while these Span's aren't null terminated, the area they're writing to has been null populated.